src/supla/sensor/AHT.h

```diff
@@ -0,0 +1,145 @@
+Skip to content
+Search or jump to…
+Pull requests
+Issues
+Marketplace
+Explore
+ 
+@milanos 
+milanos
+/
+Supla_ATH_sensor
+Public
+Code
+Issues
+Pull requests
+Actions
+Projects
+Wiki
+Security
+Insights
+Settings
+Supla_ATH_sensor/SuplaDevice/src/supla/sensor/AHT.h
+@milanos
+milanos Create AHT.h
+Latest commit cee499f 15 minutes ago
+ History
+ 1 contributor
+105 lines (88 sloc)  2.47 KB
+
+/*
+ Copyright (C) AC SOFTWARE SP. Z O.O.
+ This program is free software; you can redistribute it and/or
+ modify it under the terms of the GNU General Public License
+ as published by the Free Software Foundation; either version 2
+ of the License, or (at your option) any later version.
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+*/
+
+#ifndef SRC_SUPLA_SENSOR_AHT_H_
+#define SRC_SUPLA_SENSOR_AHT_H_
+
+#include <Adafruit_AHTX0.h>
+#include "therm_hygro_meter.h"
+
+namespace Supla {
+namespace Sensor {
+class AHT : public ThermHygroMeter {
+ public:
+	AHT(){
+    aht.begin();
+    delay(100);
+    retryCountTemp = 0;
+    retryCountHumi = 0;
+    lastValidTemp = TEMPERATURE_NOT_AVAILABLE;
+    lastValidHumi = HUMIDITY_NOT_AVAILABLE;
```

Comment:
it is better to initialize class member variables in place of their definition. I.e. instead of line 58, change line 121 to:
int8_t retryCountTemp = 0;